File: src/main/java/com/catboyranch/roborancher/commands/CRule.java
package com.catboyranch.roborancher.commands;

import com.catboyranch.roborancher.RoboRancher;
import com.catboyranch.roborancher.Server;
import com.catboyranch.roborancher.managers.RuleManager;
import com.catboyranch.roborancher.utils.RoleType;
import com.catboyranch.roborancher.utils.CommandArgument;
import com.catboyranch.roborancher.utils.Utils;
import net.dv8tion.jda.api.EmbedBuilder;
import net.dv8tion.jda.api.MessageBuilder;
import net.dv8tion.jda.api.events.message.MessageReceivedEvent;

public class CRule extends CommandBase{

    public CRule(RoboRancher rancher) {
        super(rancher);
        name = "rule";
        description = "Show a specific rule";
        allowedRoles = Utils.asArray(RoleType.ADMIN, RoleType.MODERATOR, RoleType.MEMBER);
    }

    @Override
    public void run(String cmd, Server server, MessageReceivedEvent event, CommandResult result, CommandArgument... args) {
        if(args.length == 0) {
            result.error("Please add an index for whichever rule you want to quote!");
            return;
        }

        if(!Utils.isInteger(args[0].getText())) {
            result.error("Argument must be a number!");
            return;
        }

        int index = Integer.parseInt(args[0].getText());
        RuleManager rm = server.getRuleManager();
        if(index <= 0 || index > rm.getRules().size()) {
            result.error("Bad index!");
            return;
        }

        EmbedBuilder eb = new EmbedBuilder();
        eb.setTitle(String.format("Rule %s", index));
        eb.setFooter(String.format("Requested by %s", event.getMember().getEffectiveName()));
        index--;
        eb.setDescription(rm.getRules().get(index));
        event.getMessage().delete().queue();
        event.getChannel().sendMessage(new MessageBuilder(eb).build()).queue();
        result.successQuiet();
    }
}

File: src/main/kotlin/com/catboyranch/roborancher/commands/CRule.kt
package com.catboyranch.roborancher.commands

import com.catboyranch.roborancher.ChannelUtils
import com.catboyranch.roborancher.RoboRancher
import com.catboyranch.roborancher.RoleUtils
import com.catboyranch.roborancher.Server
import net.dv8tion.jda.api.EmbedBuilder
import net.dv8tion.jda.api.MessageBuilder
import net.dv8tion.jda.api.events.message.MessageReceivedEvent

class CRule(rancher: RoboRancher): CommandBase(rancher) {
    override val name = "rule"
    override val description = "Show a specific rule"
    override val allowedRoles = arrayOf(RoleUtils.RoleType.ADMIN, RoleUtils.RoleType.MODERATOR, RoleUtils.RoleType.MEMBER)

    override fun run(cmd: String, server: Server, event: MessageReceivedEvent, result: CommandResult, vararg args: CommandArgument) {
        if(args.isEmpty()) {
            result.error("Please add an index for whichever rule you want to quote!")
            return
        }

        val rm = server.ruleManager
        var index = args[0].text.toIntOrNull()
        if(index == null || index <= 0 || index > rm.getRules().size) {
            result.error("bad index!")
            return
        }

        val builder = EmbedBuilder()
        builder.setTitle("Rule $index")
        builder.setFooter("Requested by ${event.member?.effectiveName}")
        index--
        builder.setDescription(rm.getRules()[index])
        ChannelUtils.deleteMessage(event.message)
        event.channel.sendMessage(MessageBuilder(builder).build()).queue()
        result.successQuiet()
    }
}